### ISSUE TITLE
SVG costumes don't export

### ISSUE THREAD

**Issue by maryfries** [jmoenig/Snap--Build-Your-Own-Blocks#1890]
I want to flip an SVG costume because most face right and I need one that faces left. I know we can't edit them in snap yet, but I can't export it and edit it either.

If I right-click one and choose export, a file downloads, but it won't open:

![image](https://user-images.githubusercontent.com/6549787/31785756-3924deb0-b4d4-11e7-9752-9dabc69905c8.png)
![image](https://user-images.githubusercontent.com/6549787/31785759-3b255d20-b4d4-11e7-881a-feee71f27e41.png)
![image](https://user-images.githubusercontent.com/6549787/31785769-432b3a30-b4d4-11e7-8e90-196c184d81ac.png)

**Comment by brianharvey** [comment]
Confirmed with 4.0 and 4.1 in Chrome and Firefox with the "Abby" costumes from our costume collection.

**Comment by jmoenig** [comment]
Is this *really* confirmed, Brian? does exporting "work" correctly in other browser? Did you try editing the original "Abby" costumes from the Media folder? I can import, export and re-import the "Abby" costumes fine in Chrome. Snap does *not* alter the SVG code, it just reads and dumps it, so it's rather unlikely that this is a Snap! issue.

**Comment by jguille2** [comment]
I also can confirm this bug. You can import, export and re-import successfully. The problem is when you use directly an svg costume from Snap! costumes library (_Costumes..._ option menu).Then you can't export them with success.

So the problem is not into the import method. The bug is in the Snap! image coding.

If you import a svg, you can see Snap! stores " data:image/svg+xml;base64,PH...", but if you use a Snap! svg image, you'll see "data:image/svg+xml;utf8,%3Cs..."

I've been testing, and a solution could be change the [media import process into gui.js](https://github.com/jmoenig/Snap--Build-Your-Own-Blocks/blob/master/gui.js#L3350-3351):

Then
```javascript
img.src = 'data:image/svg+xml;utf8,' +
    encodeURIComponent(txt);
```
coult change to
```javascript
img.src = 'data:image/svg+xml;base64,' +
    window.btoa(txt);
```
I think that's all. I can send a PR to facilitate the change... but I wait for your opinions.

Joan

**Comment by brianharvey** [comment]
Thanks, Joan, for coming to my rescue!  Here Jens was thinking I'm an idiot... :)

**Comment by jmoenig** [comment]
Okay, roping @bromagosa into this discussion. Do we know whether the encoding should always be base64 or do we have to remember it for each svg individually? If it's always base64 then @jguille2 's proposed fix would totally solve it, right?

**Comment by bromagosa** [comment]
I believe @jguille2's fix should work, yep :)

**Comment by jmoenig** [comment]
Okay, great! Hey, @jguille2 , I'd love to pull your fix, would you mind issuing a pull-request? Thank you so much!

**Comment by cycomachead** [comment]
Fixed in #1893
